pkg/informant/endpoints.go

```diff
@@ -375,7 +393,22 @@ func (s *State) NotifyUpscale(ctx context.Context, newResources *api.RawResource
 		return &struct{}{}, 200, nil
 	}
 
-	newMem := uint64(newResources.Memory.Value())
+	currentId := s.agents.current.id
+	incomingId := newResources.Data.Id.AgentID
+	// This condition deals with two cases:
+	// 1: The agent that responded is legitimately an unknown agent
+	// 2: The agent that responded is using an old protocol, and thus did not respond with an AgentId.
+	//    In this case, due to Go's uninitialized variable semantics, the "AgentID" we read will be
+	//    all 0's, which will never match, since UUID's cannot be all 0's
+	if incomingId != currentId {
+		klog.Errorf(
+			"Got upscale response from agent %v, while current agent is %v",
+			incomingId, currentId,
+		)
+		return nil, 400, fmt.Errorf("Received upscale response from unknown agent %v", incomingId)
+	}
```

Comment:
This check should be done before the "no action needed" check above — if the agent's bad, we _always_ want to return 400 (not: "sometimes, if the vm-informant is in a silly goofy mood")